seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-jdbc-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/FakeSourceToJdbcIT.java

```diff
@@ -54,9 +57,11 @@ public void startPostgreSqlContainer() throws InterruptedException, ClassNotFoun
                 .withLogConsumer(new Slf4jLogConsumer(LOGGER));
         Startables.deepStart(Stream.of(psl)).join();
         LOGGER.info("PostgreSql container started");
-        Thread.sleep(5000L);
         Class.forName(psl.getDriverClassName());
-        initializeJdbcTable();
+        given().ignoreExceptions()
+            .await()
+            .atMost(5, TimeUnit.SECONDS)
+            .untilAsserted(() -> initializeJdbcTable());
```

Comment:
How about adding a minimum wait time (ideally when the initialization completes short times), and then start polling until it succeeds or eventually fails or a timeout?